bin/pygrb/pycbc_grb_trig_cluster

```diff
@@ -0,0 +1,256 @@
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""Cluster triggers from a pyGRB run
+"""
+
+from __future__ import print_function
+
+import argparse
+import os
+
+import tqdm
+
+import numpy
+
+import h5py
+
+from gwdatafind.utils import filename_metadata
+
+from pycbc import __version__
+
+__author__ = "Duncan Macleod <duncan.macleod@ligo.org>"
+
+TQDM_BAR_FORMAT = ("{desc}: |{bar}| "
+                   "{n_fmt}/{total_fmt} {unit} ({percentage:3.0f}%) "
+                   "[{elapsed} | ETA {remaining}]{postfix}")
+TQDM_KW = {
+    "ascii": " -=#",
+    "bar_format": TQDM_BAR_FORMAT,
+    "smoothing": 0.05,
+}
+
+
+# -- utilities ----------------------------------
+
+def slice_hdf5(inputfile, outfile, include, verbose=False):
+    """Create a new HDF5 file containing a slice of the network events
+
+    Here ``include`` should be an index array
+    """
+    include = numpy.array(include, copy=False)
+    if include.dtype == numpy.bool_:
+        nevents = include.sum()
+    else:
+        nevents = include.size
+
+    with h5py.File(inputfile, "r") as h5in:
+        ifos = [k for k in h5in.keys() if k != "network"]
+
+        # find which single-ifo events to keep
+        ifo_index = {
+            ifo: numpy.unique(
+                h5in["network/{}_event_id".format(ifo)][:][include],
+            ) for ifo in ifos
+        }
+
+        nsets = sum(isinstance(item, h5py.Dataset) for
+                    group in h5in.values() for
+                    item in group.values())
+        msg = "Slicing {} network events into new file".format(nevents)
+        bar = tqdm.tqdm(total=nsets, desc=msg, disable=not verbose,
+                        unit="datasets", **TQDM_KW)
+        with h5py.File(outfile, "w") as h5out:
+            for old in h5in["network"].values():
+                if isinstance(old, h5py.Dataset):
+                    h5out.create_dataset(
+                        old.name,
+                        data=old[:][include],
+                        compression=old.compression,
+                        compression_opts=old.compression_opts,
+                    )
+                    bar.update()
+            for ifo in ifos:
+                idx = ifo_index[ifo]
+                for old in h5in[ifo].values():
+                    if isinstance(old, h5py.Dataset):
+                        h5out.create_dataset(
+                            old.name,
+                            data=old[:][idx],
```

Comment:
I think idx should be include.

Reply:
@idorrington92, no, `include` is the index-array of network events to keep, `idx` here is the equivalent index-array of single-detector events to keep, so this is correct.